### PR TITLE
fix: Correcting the ways pipes connect

### PR DESCRIPTION
**Pull request by supersimple33**
I have brought the fluid pipe functionality in line with the way wires are working.

### CODE REVIEW

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java [hunk 1]
```diff
@@ -80,11 +81,16 @@ public void setPlacedBy(Level level, BlockPos blockPos, BlockState blockState, @
...
             for (var direction : Constant.Misc.DIRECTIONS) {
-                var otherBlockEntity = level.getBlockEntity(blockPos.relative(direction));
-                glassPipe.getConnections()[direction.ordinal()] = (otherBlockEntity instanceof Pipe pipe && pipe.canConnect(direction.getOpposite())) || FluidUtil.canAccessFluid(level, blockPos.relative(direction), direction);
+                changed |= glassPipe.getConnections()[direction.ordinal()] = glassPipe.canConnect(direction) && FluidUtil.canAccessFluid(level, blockPos.relative(direction), direction);
+            }
+            if (changed) {
+                glassPipe.setChanged();
             }
-            level.updateNeighborsAt(blockPos, blockState.getBlock());
+            level.sendBlockUpdated(blockPos, blockState, blockState, Block.UPDATE_IMMEDIATE);
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java:93]: Could this be moved into the `if (changed)` block? (Do still we need to update the block if no connections were made?)

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java [hunk 2]
```diff
@@ -125,14 +131,21 @@ public InteractionResult use(BlockState blockState, Level level, BlockPos blockP
     @Override
     public void neighborChanged(BlockState blockState, Level level, BlockPos blockPos, Block block, BlockPos fromPos, boolean notify) {
         super.neighborChanged(blockState, level, blockPos, block, fromPos, notify);
-        var neighborState = level.getBlockState(fromPos);
-        var delta = fromPos.subtract(blockPos);
-        var direction = Direction.fromDelta(delta.getX(), delta.getY(), delta.getZ());
+        System.out.println("neighborChanged");
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java:134]: Some leftover debug logging

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java [hunk 3]
```diff
@@ -125,14 +131,21 @@ public InteractionResult use(BlockState blockState, Level level, BlockPos blockP
...
+            if (direction != null) {
+                if (!level.isClientSide
+                        && glassPipe.getConnections()[direction.ordinal()]
+                        != (glassPipe.getConnections()[direction.ordinal()]
+                        = glassPipe.canConnect(direction) && FluidUtil.canAccessFluid(level, fromPos, direction))
+                ) {
+                    level.sendBlockUpdated(blockPos, blockState, blockState, Block.UPDATE_IMMEDIATE);
+                }
+            }
+            glassPipe.setChanged();
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java:148]: Same as the other `sendBlockUpdated`/`setChanged` pair, but the other way around.

**Review by marcus8448 — CHANGES_REQUESTED** [review]
Thanks for the patch - just some minor things.

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java (hunk 1 above)
Comment (supersimple33) [reply src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java]: Yeah that makes sense. Done!

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java (hunk 2 above)
Comment (supersimple33) [reply src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java]: Removed

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java (hunk 3 above)
Comment (supersimple33) [reply src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java]: Done

File: src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java (hunk 1 above)
Comment (supersimple33) [reply src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java]: Also I made this change to wires as well

**Review by marcus8448 — APPROVED** [review]
Thanks!

**Pull request merged**